src/gt4py/next/ffront/decorator.py

```diff
@@ -411,138 +382,83 @@ def from_function(
         operator_node_cls: type[OperatorNodeT] = foast.FieldOperator,
         operator_attributes: Optional[dict[str, Any]] = None,
     ) -> FieldOperator[OperatorNodeT]:
-        operator_attributes = operator_attributes or {}
-
-        source_def = SourceDefinition.from_function(definition)
-        closure_vars = get_closure_vars_from_function(definition)
-        annotations = typing.get_type_hints(definition)
-        foast_definition_node = FieldOperatorParser.apply(source_def, closure_vars, annotations)
-        loc = foast_definition_node.location
-        operator_attribute_nodes = {
-            key: foast.Constant(value=value, type=type_translation.from_value(value), location=loc)
-            for key, value in operator_attributes.items()
-        }
-        untyped_foast_node = operator_node_cls(
-            id=foast_definition_node.id,
-            definition=foast_definition_node,
-            location=loc,
-            **operator_attribute_nodes,
-        )
-        foast_node = FieldOperatorTypeDeduction.apply(untyped_foast_node)
         return cls(
-            foast_node=foast_node,
-            closure_vars=closure_vars,
-            definition=definition,
+            definition_stage=ffront_stages.FieldOperatorDefinition(
+                definition=definition,
+                grid_type=grid_type,
+                node_class=operator_node_cls,
+                attributes=operator_attributes or {},
+            ),
             backend=backend,
-            grid_type=grid_type,
-            operator_attributes=operator_attributes,
         )
 
+    def __post_init__(self):
+        """This ensures that DSL linting occurs at decoration time."""
+        _ = self.foast_stage
```

Comment:
maybe add a remark that we want to get rid of this hard-coded post_init linting in a next step